### DISCUSSION TITLE
how to export a single layer or several layers to png?

### DISCUSSION THREAD

**Discussion by zhangyang30003** [mozman/ezdxf#444] [Q&A]
heeeeeeeeeeeeeeeeelp

**Comment by mozman** [comment]
Added an additional key-word argument `filter_func` to the `qsave()` function. 

Export only the entities on layer "example":

```Python
from ezdxf import recover
from ezdxf.addons.drawing import matplotlib

def my_filter(e: "DXFGraphic") -> bool:
    # return True to draw entity e or False to ignore it
    return e.dxf.layer == "example"

# Exception handling left out for compactness:
doc, auditor = recover.readfile("your.dxf")
if not auditor.has_errors:
    matplotlib.qsave(doc.modelspace(), "your.png", filter_func=my_filter)

```

See also docs for the drawing add-on: https://ezdxf.mozman.at/docs/addons/drawing.html

**Reply by zhangyang30003** [reply]
thanks for you reply,

i tried the example,
but thers is an error "qsave() got an unexpected keyword argument 'filter_func' ".

i'm using the 0.16.3 version, maybe 'filter_func' is only for 0.17 version?
where can i get the new version?

thanks again for your help :)

**Reply by zhangyang30003** [reply]
emm.
i found the mistake..
matplotlib.qsave is part of ezdxf.addons.drawing ,
but not matplotlib,,,

**Reply by mozman** [reply]
Official release is in v0.17, this will take a while

The feature will be available in the next bugfix release 0.16.4 in 2-3 weeks, with the risk of an API change until the official release.

Next beta release v0.16.4b0 at the coming weekend.

... immediately if you are an experienced developer:

Install from github or build from source: https://ezdxf.mozman.at/docs/setup.html#installation-from-github
